src/ui/ControlDescription.tsx

```diff
@@ -0,0 +1,175 @@
+            className={classes.apiKeyInput}
+            marginBottom="1em"
+          />
+
+          {context.endpoints.length !== 0 && (
+              <Text mt={2} color="gray.700" marginBottom="1em">
+                 Click hostname to select endpoints, then click `Describe Endpoints`
+              </Text>
+            )}
+          {context.endpoints.length === 0 && (
+            <Text mt={2} color="gray.700" marginBottom="1em">
+              No endpoints found. Activate endpoints in browser to start describing them.
+            </Text>
+          )}
+            <Accordion allowMultiple>
+              {Object.entries(endpointsByHost).map(([host, endpoints]) => (
+                <AccordionItem key={host}>
+                  <h2>
+                    <AccordionButton>
+                      <Box flex="1" textAlign="left">{host}</Box>
+                      <AccordionIcon />
+                    </AccordionButton>
+                  </h2>
+                  <AccordionPanel pb={4}>
+                    <div className={classes.wrapper}>
+                      <CheckboxGroup
+                        colorScheme="green"
+                        value={Array.from(context.selectedEndpoints)}
+                        onChange={(values) =>
+                          context.setSelectedEndpoints(new Set(values.map(value => String(value))))
+                        }
+                      >
+                        <VStack align="stretch" spacing={4}>
+                          {endpoints.map((endpoint) => {
+                            const endpointKey = `${endpoint.host}${endpoint.pathname}`;
+                            const tokenCount = context.endpointTokenCounts[endpointKey] || 0;
+                            return (
+                              <Checkbox
+                                key={`${endpoint.host}${endpoint.pathname}`}
+                                value={`${endpoint.host}${endpoint.pathname}`}
+                                onChange={() => onCheckboxChange(`${endpoint.host}${endpoint.pathname}`)}
+                              >
+                                {endpoint.pathname} | Tokens: {tokenCount}
+                              </Checkbox>
+                            );
+                          })}
+                        </VStack>
+                      </CheckboxGroup>
+                    </div>
+                  </AccordionPanel>
+                </AccordionItem>
+              ))}
+            </Accordion>
+          <Button
+            mt={4}
+            colorScheme="blue"
+            isDisabled={(context.selectedEndpoints.size === 0 || context.descriptionsLoading)}
+            onClick={handleDescribeEndpoints}
+          >
+            Describe Endpoints
+          </Button>
+          {context.descriptionsLoading && (
+            <Text mt={2} color="gray.700" marginBottom="1em">
+              Describing endpoints... 
+            </Text>
+          )}
```

Comment:
Similar to this, could we have a temporary 'success' message pop up when it succeeds?

Reply:
yes, in new commit :)